### PR TITLE
ENYO-2709: Added support for sideways-facing tooltips

### PR DESCRIPTION
**Pull request by Djspaceg**
Tooltip nub also got a makeover in all cases, and now looks more like the original spec.

Enyo-DCO-1.1-Signed-off-by: Blake Stephens <blake.stephens@lge.com>

### CODE REVIEW

File: src/Tooltip/Tooltip.js [hunk 1]
```diff
@@ -314,7 +319,7 @@ module.exports = kind(
 	* @private
 	*/
 	positionChanged:function () {
-		Popup.prototype.positionChanged.apply(this, arguments);
+		Popup.prototype.positionChanged && Popup.prototype.positionChanged.apply(this, arguments);
```
Comment (Djspaceg) [inline src/Tooltip/Tooltip.js]: `positionChanged` was behaving strangely when the value was in a binding. Them _seemed_ to remedy the problem.

File: src/Tooltip/Tooltip.js [hunk 2]
```diff
@@ -281,6 +284,8 @@ module.exports = kind(
 		if (this.contentUpperCase !== null) this.uppercase = this.contentUpperCase;
 
 		this.contentChanged();
+
+		this.$.point.setAttribute('viewBox', '0 0 3 5');
```
Comment (viodragon2) [inline src/Tooltip/Tooltip.js]: can we put this inside the components block instead? 
e.g. `{name: 'point', kind: Control,  ... , attributes: {viewBox: '0 0 3 5'}`

File: src/Tooltip/Tooltip.js (hunk 1 above)
Comment (viodragon2) [inline src/Tooltip/Tooltip.js]: Surprised this never brought any problems before. There's no `positionChanged` in `enyo/Popup`. We should get rid of this line instead.

**Comment by viodragon2** [comment]
There's no documentation on how to use sideway tooltip. I think editing `position` property would suffice.

**Comment by viodragon2** [comment]
Also, `top` and `bottom` seemed a bit confusing to me. They represent where the tail resides in respect to the tooltip instead of direction of where tooltip is facing.

One last thing I want to mention is that it doesn't flip when the sideway tooltips go over the viewport of window like the `above` and `below` case. I don't think it's necessary, but it's worth mentioning in somewhere as part of the doc if we're not going to address those cases. (`below` position doesn't properly flip either. We should fix them all together if we do.)

File: src/Tooltip/Tooltip.js (hunk 2 above)
Comment (Djspaceg) [inline src/Tooltip/Tooltip.js]: Of course! I didn't know you could interact with attributes that way. Will do.

**Comment by Djspaceg** [comment]
All issues including documentation and inverted top/bottom addressed. (As well as a new bug found, and corrected. "auto" wasn't working as expected if chosen after a different position.)

**Pull request merged**